File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/sub_categoria.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

public partial class sub_categoria
{
    [Key]
    public int id_sub_categoria { get; set; }

    [Required(ErrorMessage = "El nombre de la subcategoría es obligatorio.")]
    [StringLength(255, ErrorMessage = "El nombre de la subcategoría no puede exceder los 255 caracteres.")]
    public string nombre_sub_categoria { get; set; } = null!;

    [Required(ErrorMessage = "El ID de la categoría es obligatorio.")]
    public int id_categoria { get; set; }

    [ForeignKey("id_categoria")]
    [InverseProperty("sub_categoria")]
    public virtual categoria id_categoriaNavigation { get; set; } = null!;

    [InverseProperty("id_sub_categoriaNavigation")]
    public virtual ICollection<producto> productos { get; set; } = new List<producto>();
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/sub_categoria.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

// Esquema del modelo SubCategoria
const subCategoriaSchema = new Schema({
    nombre_sub_categoria: {
        type: String,
        required: [true, 'El nombre de la subcategoría es obligatorio.'],
        maxlength: [255, 'El nombre de la subcategoría no puede exceder los 255 caracteres.']
    },
    id_categoria: {
        type: Schema.Types.ObjectId,
        required: [true, 'El ID de la categoría es obligatorio.'],
        ref: 'categoria' // Asegúrate de tener el modelo "Categoria" definido
    },
    productos: [{
        type: Schema.Types.ObjectId,
        ref: 'producto' // Asegúrate de tener el modelo "Producto" definido
    }]
});

// Crea el modelo SubCategoria
const SubCategoria = mongoose.model('SubCategoria', subCategoriaSchema);

module.exports = SubCategoria;
